pages/docs/tooling/tooling-aste.md

```diff
@@ -0,0 +1,141 @@
+permalink: tooling-aste.html
+keywords: tooling, aste
+---
+
+## Motivation
+
+ASTE is a tool which can be used to understand the dynamics of preCICE, how the data is goes in and out. It consist of C++ and Python modules.
+
+## Installation
+
+Please make sure that dependencies are installed.
+
+### C++ modules
+
+#### Required
+
+- preCICE
+- MPI
+- CMake
+
+##### Optional
+
+- METIS
+
+### Python modules
+
+#### Required
+
+- NumPy
+- vtk (Visualization Toolkit)
+
+### Building
+
+```bash
+mkdir build
+cd build
+cmake ..
+make
+```
+
+If precice is not installed in `$PRECICE_ROOT/build` do `cmake -DCMAKE_LIBRARY_PATH=$PRECICE_INSTALL_DIR ..` with the correct installation directory.
+
+After building, add the `build/` directory to your `PATH`.
+
+### on SuperMUC
+
+Initialize environment, e.g., put in your `.bashrc`
+
+```
+# METIS for aste
+module load metis
+export METIS_DIR=$METIS_BASE # so that cmake finds metis
+export CPLUS_INCLUDE_PATH="$METIS_BASE/include:$CPLUS_INCLUDE_PATH"
+```
+
+and execute cmake like `cmake -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ ..` and `make`.
+
+## Usage
+
+## C++ modules
+
+### preciceMap
+
+The main tool used for mapping.
+
+| Flag      | Explanation                                            |
+| --------- | ------------------------------------------------------ |
+| -v        | Enables verbose output                                 |
+| -c        | To specify preCICE config file (default="precice.xml") |
+| -p        | Participant name either A or B                         |
+| --runName | Name of run (default="")                               |
+| --data    | Name of data array to be mapped                        |
+| --mesh    | Mesh prefix                                            |
+| --output  | Output filename (default=output)                       |
+| --vector  | A bool switch to specify vector data (default=False)   |
+
+Sample usage is
+
+```bash
+preciceMap -v -p A --mesh fine_mesh/fine_mesh --data "x + y" 
+```
+
+## Python modules
+
+### vtk_calculator.py
+
+Reads a mesh as either `.vtk` or `.vtu` and evaluates a function given by `-f` on it. Using the `-d` flag, it can compute the difference between the mesh values and the values of the analytical solution (usually applied after a mapping).
+
+| Flag | Explanation                                                                         |
+| ---- | ----------------------------------------------------------------------------------- |
+| -o   | Output meshname if it is not given data appended to input mesh                      |
+| -t   | Tag for the data created (default="MyScalar")                                       |
+| -it  | The tag for input data. Used in diff mode. If not given tag is used.                |
+| -l   | Logging level (default="INFO")                                                      |
+| -d   | Calculate the difference to present data.                                           |
```

Comment:
What do you mean by "present data"? Clarify / rephrase.

Reply:
`Calculates the difference between --diffdata and given function. ` probably more clear